sites/gamestop.py

```diff
@@ -0,0 +1,168 @@
+from selenium import webdriver
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait as wait
+from webdriver_manager.chrome import ChromeDriverManager
+from chromedriver_py import binary_path as driver_path
+from utils import random_delay, send_webhook
+import string, random, settings, time, re
+
+class GameStop:
+    def __init__(self, task_id, status_signal, image_signal, product, profile, proxy, monitor_delay, error_delay, max_price):
+        self.task_id, self.status_signal, self.image_signal, self.product, self.profile, self.monitor_delay, self.error_delay, self.max_price = task_id, status_signal, image_signal, product, profile, float(
+            monitor_delay), float(error_delay), max_price
+
+        starting_msg = "Starting GameStop"
+        self.browser = self.init_driver()
+        self.product_image = None
+
+        self.SHORT_TIMEOUT = 5
+        self.LONG_TIMEOUT = 15
+
+        if settings.dont_buy:
+            starting_msg = "Starting GameStop in dev mode; will not actually checkout."
+
+        self.status_signal.emit(self.create_msg(starting_msg, "normal"))
+        self.login()
+        self.monitor()
+        self.add_to_cart()
+        self.submit_billing()
+        self.submit_order()
+
+
+
+    def create_msg(self, msg, status):
+        return {"msg": msg, "status": status}
+
+
+    # https://stackoverflow.com/questions/33225947/can-a-website-detect-when-you-are-using-selenium-with-chromedriver
+    def change_driver(self, loc):
+        print(loc)
+        fin = open(loc, 'rb')
+        data = fin.read()
+        val = "$cdc_" + "".join(random.choices(string.ascii_letters + string.digits, k=22)) + "_"
+
+        result = re.search(b"[$]cdc_[a-zA-Z0-9]{22}_", data)
+
+        if result is not None:
+            self.status_signal.emit(self.create_msg("Changing value in Chromedriver", "normal"))
+            data = data.replace(result.group(0), val.encode())
+            fin.close()
+            fin = open(loc, 'wb')
+            fin.truncate()
+            fin.write(data)
+            fin.close()
+        else:
+            fin.close()
+
+
+    def init_driver(self):
+        driver_manager = ChromeDriverManager()
+        driver_manager.install()
+        # self.change_driver(driver_path)
+        browser = webdriver.Chrome(driver_path)
+
+        browser.execute_cdp_cmd("Page.addScriptToEvaluateOnNewDocument", {
+            "source": """
+                  Object.defineProperty(navigator, 'webdriver', {
+                   get: () => undefined
+                  })
+                """
+        })
+
+        return browser
+
+
+    def login(self):
+        my_username = "username"
+        my_password = "password"
```

Comment:
We need to add this to the settings page for login credentials per task.

Reply:
Agreed, needs to be added to GUI.